src/main/java/com/googleinterns/zoomtube/servlets/IconFeedbackServlet.java

```diff
@@ -62,6 +67,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(createEntityFromRequest(request));
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> getRequestError = validateGetRequest(request);
+    if (getRequestError.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, getRequestError.get());
+      return;
+    }
+
+    List<IconFeedback> lectures = getIconFeedbacks(request);
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /** Returns IconFeedbacks stored in the database. */
+  private List<IconFeedback> getIconFeedbacks(HttpServletRequest request) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
```

Comment:
I think it might be cleaner to get the parameter and make the key in doGet, and pass the lecture key to this function

Reply:
My reasoning for putting it here is that I don't need the `lecture` in `doGet()`, keeping it here would encapsulate `lecture` to this method. Also, `lecture` is closely related to creating the filter and it would make sense.

I also see why placing `lecture` in `doGet()` would make `getIconFeedbacks()` cleaner as it only has to retrieve the data from the database.

Reply:
I think you can also make it a bit cleaner by separating the code for getting the query into its own helper method.

Reply:
I think it is good for now, is that ok?

Reply:
True... I personally don't like passing this giant `request` object when all that you care about is the single `lecture` parameter.  Either way is a trade-off tho, so up to u

Reply:
Ah, sounds good